meetings/2020-07/july-20.md

```diff
@@ -0,0 +1,543 @@
+
+RBN: That is feasible, but there are some caveats as well. Class expressions and default class exports can be unnamed. I don’t believe your question was specifically around whether or not we should use “this”, but there are going to be cases where there is no name being referenced.
+
+MM: This part, I have no argument with, I think it's the right thing. I'm saying the class name could be available inside the static block even though outside the class as a whole, it’s temporal deadzone until the class is initialized.
+
+RBN: That would likely be the case. Following along with how class definition works, the name would exist, but it’s not actually bound with a value until the definition has completed evaluation, because of how the double-binding works.
+
+MM: I like that double-binding, is that in the proposal now?
+
+RBN: Double binding already exists. I'm trying to find out how to fit it properly into the spec text. Generally the evaluation of these statements would happen after the outer definition has been evaluated and TDZ is bound, so I’m still trying ???.
+
+MM: So, that part I very much object to. I think the class name outside the class should be in TDZ until the class is initialized.
+
+RBN: No contention about that at all. I agree that’s the case.
+
+MM: Oh! Okay then I misunderstood something.
+
+RBN: That's how it currently works.  I'm trying to say the class will be TDZ outside the class definition until the class is fully initialized. Which should align with how static fields are initialized.
+
+MM: Great. I very much support this for stage 2.
+
+SYG: I really do like the multi-step static initialization. I think that alone is enough motivation for something like this. My one comment right now is - and as folks have pointed out this is not a stage 2 blocker right now - I’m unconvinced by the special `var` treatment. I would like to think of this as not a function boundary but a block that runs. So, to have vars not escape the static block seems weird to me. I'd like to understand the motivation for that better, for Stage 3.
+
+RBN: My question would be, how would you imagine that `var` would work in that case? Would it lift out of static block into the surrounding function scope?
+
+SYG: Yes
+
+RBN: I’ll be honest, if I could’ve written this as `static(){`, then I would have, but that's already legal JS 
+Because it has special semantics around how `this` works, because otherwise who knows what `this` would be referring to, and then we have the complexity around what happens with return, await, yield. I saw on IRC that there’s some discussion about the class access constructions, that one, I may eventually bring back, but that one had some strange things around evaluation, since its main motivation was around private static and subclassing.
```

Comment:
```suggestion
Because it has special semantics around how `this` works, because otherwise who knows what `this` would be referring to, and then we have the complexity around what happens with return, await, yield. I saw on IRC that there’s some discussion about the class access expressions, that one, I may eventually bring back, but that one had some strange things around evaluation, since its main motivation was around private static and subclassing.
```